Moved GRAVITY_* constants to Item model class

Quix0r opened the pull request:
According to @MrPetovan constants in `boot.php` that are not used in configuration files can be safely moved to proper classes. This PR moves all `GRAVITY_*` constants to `Friendica\Model\Item` class.

Quix0r committed:
Rewrite:
- moved constants GRAVITY_* from boot.php to Friendica\Model\Item
- also rewrote some array initialization:

From:
````
<?php
$arr = [];
$arr['foo'] = "FOO";
````

To:
````
<?php
$arr['foo'] = "FOO";
````
- added a few type-hints

Quix0r committed:
Changes:
- unified class import, we had these both:

````
use Friendica\Model\Item as ModelItem;
use Friendica\Object\Post as PostObject;
````

In first version the namespace was prefixing class name, in second suffixing it.

Quix0r committed:
Fixed:
- Content\Item versus Model\Item is really confusing!

Quix0r committed:
Changes:
- Content\Item versus Model\Item solved (?)

Quix0r committed:
Rewrites:
From:
````
<?php
$arr = [];
$arr['foo'] = "FOO";
````

To:
````
<?php
$arr['foo'] = "FOO";
````

MrPetovan commented on mod/oexchange.php:
```diff
@@ -109,7 +109,7 @@ function oexchange_content(App $a) {
 	$tags        = !empty($_REQUEST['tags'])        ? trim($_REQUEST['tags'])        : '';
 
-	$s = \Friendica\Content\Text\BBCode::embedURL($url, true, $title, $description, $tags);
+	$s = BBCode::embedURL($url, true, $title, $description, $tags);
```
The related `use Friendica\Content\Text\BBCode;` statement seems to be missing.

MrPetovan commented on mod/oexchange.php:
```diff
@@ -119,9 +119,9 @@ function oexchange_content(App $a) {
 
 	$post['profile_uid'] = local_user();
 	$post['return'] = '/oexchange/done';
-	$post['body'] = Friendica\Content\Text\HTML::toBBCode($s);
```
The related `use Friendica\Content\Text\HTML;` statement seems to be missing.

MrPetovan commented on mod/tagger.php:
```diff
@@ -116,41 +116,41 @@ function tagger_content(App $a) {
+		'author-name'   => $contact['name'],
+		'author-link'   => $contact['url'],
+		'author-avatar' => $contact['thumb'],
+		'body'          => sprintf( $bodyverb, $ulink, $alink, $plink, $termlink),
```
Code standards.
```suggestion
		'body'          => sprintf($bodyverb, $ulink, $alink, $plink, $termlink),
```

MrPetovan commented on view/theme/frio/theme.php:
```diff
@@ -33,7 +33,9 @@
 use Friendica\Database\DBA;
 use Friendica\DI;
 use Friendica\Model;
```
Is this still needed?

Quix0r replied:
I don't know. But that the fact that themes can have access to database (circumventing the controller) is scary.

annando commented:
Overall the usage of an IDE with built-in syntax checker would help here. Especially with this huge amount of files, the possibility of syntax errors is immense.

It would be much better, not to have performed multiple steps (exchanging the constants and code reformatting) in one huge PR. Also it would had been better to keep the old constants for some time and just deprecating them, since they are also used in the addons.

Quix0r committed:
Added missing `use` statement, thanks to @MrPetovan

Quix0r committed:
Added missing `use` statement (again), thanks to @MrPetovan

Yes, we need more unit-testing! Not only to rely on your IDE that it will locate
it for you.

Quix0r commented:
> Overall the usage of an IDE with built-in syntax checker would help here. Especially with this huge amount of files, the possibility of syntax errors is immense.

I strongly disagree here. Better more unit-testing which would "on-the-fly" test this for us as a by-product.

Quix0r committed:
Ops, code-convention not followed

Quix0r committed:
Merge branch 'develop' into rewrite/gravity-constants

Quix0r commented:
And for addons, I have also made a branch ready for PR.
